kernel/mem_slab.c

```diff
@@ -206,12 +206,19 @@ int k_mem_slab_init(struct k_mem_slab *slab, void *buffer,
 
 static bool slab_ptr_is_good(struct k_mem_slab *slab, const void *ptr)
 {
+#ifdef CONFIG_MEM_SLAB_POINTER_VALIDATE
```

Comment:
Style: returning "Yes the pointer is good!" when it's not good is sort of a violation of the spirit of the API naming here.   Consider just wrapping the calls (I count only two) with `IS_ENABLED(CONFIG_MEM_SLAB_POINTER_VALIDATE)` (or fold it into the `__ASSERT()` test I guess).